Fixed code for error handling and structure in shaku_rocket

BennyPLS opened the pull request:
Converted from map_failure to map_error and into_outcome to a match.

AzureMarker commented:
Hey, thanks for the PR. Was there a change that caused this code to break, or is this just for style?

BennyPLS commented:
Sorry for my poor writing. I am not a great writer or explainer, I just modified 2 parts.

- into_outcome

`into_outcome` I couldn't find an equivalent for the last rocket version I just modified the code using a match statement.

- into_failure

The Outcome variant was renamed to Error. into_failure -> into_error

It's my first pull request I am not very experienced, Please review the code for my use case and test pass.

AzureMarker commented:
I see, the library was written against the rocket 0.5.0 release candidates (e.g. 0.5.0-rc1), but the dependency semantics means the library also technically works with 0.5.0. In rocket 0.5.0 some things were changed, which causes compile time errors.
